### PR TITLE
Smoothened transition between base activities.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes: #1825

Changes: Smoothened animation while switching activities.

Screenshots for the change: 

anim-opt

### CODE REVIEW

**Comment by yashLadha** [comment]
Done @sridharjajoo . Please Update your review.

**Comment by yashLadha** [comment]
Ping @mariobehling

**Comment by isuruAb** [comment]
@abishekvashok  please review

File: app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java [hunk 1]
```diff
@@ -140,8 +140,8 @@ public void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
         ActivitySwitchHelper.setContext(this);
         parentLayout.setBackgroundColor(getBackgroundColor());
-        this.overridePendingTransition(R.anim.right_to_left,
-                               R.anim.left_to_right);parentLayout.setBackgroundColor(getBackgroundColor());
+        overridePendingTransition(R.anim.right_to_left,
+                R.anim.left_to_right);parentLayout.setBackgroundColor(getBackgroundColor());
```
Comment (abishekvashok) [inline app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java]: There's a `;` in between. Can you make the other line to appear on the next line?

File: app/src/main/java/org/fossasia/phimpme/base/BaseActivity.java [hunk 2]
```diff
@@ -132,13 +132,25 @@ public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
         if (item.getItemId() != getNavigationMenuItemId()) {
             switch (item.getItemId()) {
                 case R.id.navigation_camera:
-                    startActivity(new Intent(this, CameraActivity.class).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP));
+                    Intent cameraIntent = new Intent(this, CameraActivity.class);
+                    startActivity(cameraIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP));
+                    finish();
+                    overridePendingTransition(R.anim.left_to_right,
```
Comment (abishekvashok) [inline app/src/main/java/org/fossasia/phimpme/base/BaseActivity.java]: Can finish and this line be moved out of the switch?

**Comment by yashLadha** [comment]
Done @abishekvashok

**Comment by raj10071997** [comment]
@yashLadha  I see that in some places you have initialized the variable outside the function calling and then passed them in the argument. Like this - 
` ` `
  Intent cameraIntent = new Intent(this, CameraActivity.class);
            startActivity(cameraIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP));
` ` `
Actually I don't know if this improves the performance. But in my view, java is pass by value, so if you initialize the variable inside the function call or outside the function call and then pass it as parameter, what difference does it create? Because whatever will be the case first the parameters will be resolved  and memory will be assigned to them first and then their memory reference will be passed as value. Actually I don't know whether writing the code like this will improve the performance but this was a genuine doubt of mine, so I asked the question?

**Comment by yashLadha** [comment]
Actually the animation currently is sort of periodic and not according to the previous state. I am planning to change the logic in other PR about including the animation according to a specific logic. @raj10071997

**Pull request merged**